gruntwork-install

```diff
@@ -202,40 +203,45 @@ function get_os_arch_gox_format {
 # We release binaries with the name following the format <NAME>_<OS>_<ARCH> (e.g. foo_linux_amd64). Given the NAME of
 # a binary, this function adds the proper OS and ARCH to it for the current OS.
 function determine_binary_name {
-  local readonly binary_name="$1"
-  local readonly os_name=$(get_os_name)
-  local readonly os_arch=$(get_os_arch_gox_format)
+  local -r binary_name="$1"
+  local -r os_name=$(get_os_name)
+  local -r os_arch=$(get_os_arch_gox_format)
   echo "${binary_name}_${os_name}_${os_arch}"
 }
 
 # Take in a key-value pair of the format key=value and convert it to the format that Gruntwork bash scripts expect:
 # --key value
 function convert_module_params_format {
-  local readonly key_value_pair="$1"
-  local readonly key="${key_value_pair%%=*}"
-  local readonly val="${key_value_pair#*=}"
-  echo "--${key} \"${val}\""
+  local -r key_value_pair="$1"
+  local -r key="${key_value_pair%%=*}"
+  local -r val="${key_value_pair#*=}"
+  local -ra params=( "--${key}" "${val}")
+  
+  # Print the value out as something that can be evaled back into an array
+  # Note: this prints something like '([0]="key_name" [1]="value1 value2 value3")'
```

Comment:
Is the issue related specifically to printing? If so, we _could_ extract this logic out of this function and inline it in the location where it's being used so that `key` and `val` are separate.

Reply:
Yes, it's the printing (since bash doesn't actually "return" values). I just pushed a change that inlines this and it's honestly a lot cleaner.

Reply:
Yea... That's much better!